### PR TITLE
crowdin-cli: init at 3.6.4

### PR DESCRIPTION
**Pull request by DamienCassou**
###### Motivation for this change

Add crowdin-cli

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by DamienCassou** [comment]
Result of `nixpkgs-review pr 133178` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>crowdin-cli</li>
  </ul>
</details>

File: pkgs/tools/text/crowdin-cli/default.nix [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    install -D crowdin_completion $out/share/bash-completion/completions/crowdin
+
+    mkdir $out/bin
+
+    cat > $out/bin/crowdin << EOF
+    #! ${stdenv.shell} -e
+    exec ${jre}/bin/java -jar '$out/lib/crowdin-cli.jar' "\$@"
+    EOF
+
+    chmod +x $out/bin/crowdin
```
Comment (SuperSandro2000) [inline pkgs/tools/text/crowdin-cli/default.nix:43]: This could be easily replaced by makeWrapper.

File: pkgs/tools/text/crowdin-cli/default.nix [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+      gawk
+    gnugrep
+  git
+  ];
+
+  installPhase = ''
+    runHook preInstall
+
+    install -D crowdin-cli.jar $out/lib/crowdin-cli.jar
+    install -D crowdin_completion $out/share/bash-completion/completions/crowdin
```
Comment (SuperSandro2000) [inline pkgs/tools/text/crowdin-cli/default.nix:34]: Do we have zsh or fish?

Please use installShellCompletion.

File: pkgs/tools/text/crowdin-cli/default.nix [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  propagatedBuildInputs = [
+      gawk
+    gnugrep
+  git
+  ];
+
+  installPhase = ''
+    runHook preInstall
+
+    install -D crowdin-cli.jar $out/lib/crowdin-cli.jar
```
Comment (SuperSandro2000) [inline pkgs/tools/text/crowdin-cli/default.nix:33]: Not sure if we should put this into a subdirectory under lib.

File: pkgs/tools/text/crowdin-cli/default.nix [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    sha256 = "123mv0s1jppidmwsvr8a6f8429xmpskxmnv4p8jpnfa9zrw86aaw";
+  };
+
+  nativeBuildInputs = [ unzip ];
+
+  propagatedBuildInputs = [
+      gawk
+    gnugrep
+  git
+  ];
```
Comment (SuperSandro2000) [inline pkgs/tools/text/crowdin-cli/default.nix:28]:
```suggestion
  propagatedBuildInputs = [
    gawk
    gnugrep
    git
  ];
```

These should be added to the PATH in makeWrapper.

File: pkgs/tools/text/crowdin-cli/default.nix [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+, gnugrep
+, jre
+, unzip
+}:
+
+let
+  pname = "crowdin-cli";
+  version = "3.6.4";
+in stdenv.mkDerivation {
+  inherit pname version;
```
Comment (SuperSandro2000) [inline pkgs/tools/text/crowdin-cli/default.nix:15]:
```suggestion
stdenv.mkDerivation rec {
  pname = "crowdin-cli";
  version = "3.6.4";
```

File: pkgs/tools/text/crowdin-cli/default.nix (hunk 1 above)
Comment (DamienCassou) [reply pkgs/tools/text/crowdin-cli/default.nix:43]: excellent suggestion. Much better!

File: pkgs/tools/text/crowdin-cli/default.nix (hunk 2 above)
Comment (DamienCassou) [reply pkgs/tools/text/crowdin-cli/default.nix:34]: > Do we have zsh or fish?

only bash is provided

> Please use installShellCompletion.

I love these helpers, thanks!

File: pkgs/tools/text/crowdin-cli/default.nix (hunk 3 above)
Comment (DamienCassou) [reply pkgs/tools/text/crowdin-cli/default.nix:33]: I looked at the `umlet` package and the jar is directly under `lib/` but I don't know if that is a good example.

File: pkgs/tools/text/crowdin-cli/default.nix (hunk 4 above)
Comment (DamienCassou) [reply pkgs/tools/text/crowdin-cli/default.nix:28]: done. Do you know a better way?

File: pkgs/tools/text/crowdin-cli/default.nix (hunk 5 above)
Comment (DamienCassou) [reply pkgs/tools/text/crowdin-cli/default.nix:15]: done. I usually avoid `rec` expression and follow recommendations from https://nix.dev/anti-patterns/language#rec-expression. Why do you prefer using `rec`?

**Comment by DamienCassou** [comment]
Thank you very much for your feedback @SuperSandro2000! It is very much appreciated. I took it all into account but left 2 questions for you.

File: pkgs/tools/text/crowdin-cli/default.nix [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+    runHook preInstall
+
+    install -D crowdin-cli.jar $out/lib/crowdin-cli.jar
+
+    installShellCompletion --cmd crowdin --bash ./crowdin_completion
+
+    makeWrapper ${jre}/bin/java $out/bin/crowdin \
+      --argv0 crowdin \
+      --add-flags "-jar $out/lib/crowdin-cli.jar" \
+      --suffix-each PATH : "${gawk}/bin ${gnugrep}/bin ${git}/bin"
```
Comment (SuperSandro2000) [inline pkgs/tools/text/crowdin-cli/default.nix:40]:
```suggestion
      --prefix PATH : ${lib.makeBinPath [ gawk gnugrep git ]}"
```

File: pkgs/tools/text/crowdin-cli/default.nix [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+    sha256 = "123mv0s1jppidmwsvr8a6f8429xmpskxmnv4p8jpnfa9zrw86aaw";
+  };
+
+  nativeBuildInputs = [ installShellFiles makeWrapper unzip ];
+
+  propagatedBuildInputs = [
+    gawk
+    gnugrep
+    git
+  ];
```
Comment (SuperSandro2000) [inline pkgs/tools/text/crowdin-cli/default.nix:28]:
```suggestion
```

File: pkgs/tools/text/crowdin-cli/default.nix (hunk 6 above)
Comment (DamienCassou) [reply pkgs/tools/text/crowdin-cli/default.nix:40]: awesome, thanks!

File: pkgs/tools/text/crowdin-cli/default.nix (hunk 7 above)
Comment (DamienCassou) [reply pkgs/tools/text/crowdin-cli/default.nix:28]: done

**Comment by DamienCassou** [comment]
Thank you again for your suggestions that I applied. There is still one question from me regarding the use of `rec` vs. `let`.

**Comment by DamienCassou** [comment]
Result of `nixpkgs-review pr 133178` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>crowdin-cli</li>
  </ul>
</details>

File: pkgs/tools/text/crowdin-cli/default.nix (hunk 5 above)
Comment (SuperSandro2000) [reply pkgs/tools/text/crowdin-cli/default.nix]: That doesn't apply to mkDerivation really. It is more useful when writing library functions.

**Pull request merged**